File: C++/SORTING/01-Bubble_Sort/bubble_sort.py
def bubbleSort(arr):
    n = len(arr)
    for i in range(n):
        swapped = False
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
                swapped = True
        if(swapped == False):
            break

if __name__ == "__main__":
    arr = list(map(int, input().split()))
    # arr = [64, 34, 1, 6, 78, 11, 12, 33, 22, 3]
    for i in range(len(arr)):
        print("%d" % arr[i], end = " ")
    print()            
    
    bubbleSort(arr)
    
    print("Sorted array: ")
    for i in range(len(arr)):
        print("%d" % arr[i], end = " ")

File: C++/SORTING/01-Bubble_Sort/bubble_sort.cpp
#include <bits/stdc++.h>
#define RUN_TIME
#define fastio() ios_base::sync_with_stdio(false)
using namespace std;

void swap(int *a, int *b) {
    int temp = *a;
    *a = *b;
    *b = temp;
}

void print(int a[], int n) {
    for(int i=0; i < n; i++) {
        cout << a[i] << " ";
    }
    cout << endl;
}

void bubbleSort(int a[], int n) {   // O(n^2)
    // logic implementation section
    int i, j;
    bool swapped;
    for(i = 0; i < n - 1; i++) {
        swapped = false;
        for(j = 0; j < n - i - 1; j++) {
            if(a[j] > a[j + 1]) {
                swap(a[j], a[j + 1]);
                swapped = true;
            }
        }
        if(!swapped) break;
    }    
}

int main() {
    fastio();

    int arr[] = { 64, 34, 25, 12, 22, 11, 90 };
    int N = sizeof(arr) / sizeof(arr[0]);
    bubbleSort(arr, N);
    cout << "Sorted array: \n";
    print(arr, N);

#ifdef RUN_TIME
    cerr << "\nTime elapsed: " << 1.0 * clock() / CLOCKS_PER_SEC << " s.\n";
#endif
    return 0;
}